File: solutions/zig/difference-of-squares/2/difference_of_squares.zig
pub fn squareOfSum(number: u64) u64 {
    var total: u64 = 0;
    for (0..number + 1) |i| {
        total += i;
    }
    return total * total;
}

pub fn sumOfSquares(number: u64) u64 {
    var total: u64 = 0;
    for (0..number + 1) |i| {
        total += i * i;
    }
    return total;
}

pub fn differenceOfSquares(number: u64) u64 {
    return squareOfSum(number) - sumOfSquares(number);
}

File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number) {
  unsigned int i;
  unsigned int sum = 0;
  for (i = 0; i <= number; i++) {
    sum += i * i;
  }
  return sum;
}

unsigned int square_of_sum(unsigned int number) {
  unsigned int i;
  unsigned int sum;
  for (i = 0; i <= number; i++) {
    sum += i;
  }
  return sum * sum;
}

unsigned int difference_of_squares(unsigned int number) {
  return square_of_sum(number) - sum_of_squares(number);
}
